xla/service/gpu/transforms/cudnn_sdpa_score_mod.h

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2025 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_SERVICE_GPU_TRANSFORMS_CUDNN_SDPA_SCORE_MOD_H_
+#define XLA_SERVICE_GPU_TRANSFORMS_CUDNN_SDPA_SCORE_MOD_H_
+
+#include "absl/functional/function_ref.h"
+#include "absl/status/status.h"
+#include "third_party/cudnn_frontend/include/cudnn_frontend.h"
+#include "xla/hlo/ir/hlo_computation.h"
+
+namespace stream_executor {
+namespace gpu {
+
+using Tensor = std::shared_ptr<cudnn_frontend::graph::Tensor_attributes>;
+using Graph = std::shared_ptr<cudnn_frontend::graph::Graph>;
+using Uid = absl::FunctionRef<int64_t()>;
+
+class ScoreModFunc {
+ public:
+  ScoreModFunc(const xla::HloComputation* fwd_comp,
+               const xla::HloComputation* bwd_comp);
+
+  std::optional<cudnn_frontend::PointwiseMode_t> GetElementwiseMode(
+      const xla::HloInstruction& instruction);
+
+  cudnn_frontend::DataType_t GetComputeDataType(const xla::PrimitiveType type);
+
+  template <xla::PrimitiveType XlaT, typename T>
+  Tensor LiteralToCudnnTensor(const xla::HloInstruction* hlo,
+                              cudnn_frontend::graph::Graph& graph);
+
+  absl::Status UpdateCudnnMap(cudnn_frontend::graph::Graph& graph,
+                              Uid next_uid);
+
+  absl::Status UpdateHloParameterToCudnnMap(
+      cudnn_frontend::graph::Graph& graph,
+      absl::flat_hash_map<const xla::HloInstruction*, Tensor>& hlo_to_cudnn,
+      const xla::HloComputation* computation, Uid next_uid);
+
+  absl::Status UpdateHloConstantToCudnnMap(
+      cudnn_frontend::graph::Graph& graph,
+      absl::flat_hash_map<const xla::HloInstruction*, Tensor>& hlo_to_cudnn,
+      const xla::HloComputation* computation);
```

Comment:
Should these be private functions? It appears they don't get called from anywhere else.